src/xunit.runner.uap/RunnerRemoteExecutionServiceUAP/RunnerRemoteExecutionServiceUAP.cs

```diff
@@ -0,0 +1,146 @@
+            // from that assembly to find, and the method from that assembly to invoke.
+            // Any additional arguments are passed as strings to the method.
+
+            if (!GetStringValueFromValueSet(message, "AssemblyName", out assemblyName) || String.IsNullOrEmpty(assemblyName))
+            {
+                returnData["Results"] = -1;
+                returnData["Log"] = $"RemoteExecuter Error: Missing assembly name";
+                return;
+            }
+
+            if (!GetStringValueFromValueSet(message, "TypeName", out typeName) || String.IsNullOrEmpty(typeName))
+            {
+                returnData["Results"] = -2;
+                returnData["Log"] = $"RemoteExecuter Error: Missing type name";
+                return;
+            }
+
+            if (!GetStringValueFromValueSet(message, "MethodName", out methodName) || String.IsNullOrEmpty(methodName))
+            {
+                returnData["Results"] = -3;
+                returnData["Log"] = $"RemoteExecuter Error: Missing method name";
+                return;
+            }
+
+            List<string> argList = new List<string>();
+            int i = 0;
+            string arg;
+            while (GetStringValueFromValueSet(message, "Arg"+i, out arg))
+            {
+                argList.Add(arg);
+                i++;
+            }
+            additionalArgs = argList.ToArray();
+
+            // Load the specified assembly, type, and method, then invoke the method.
+            // The program's exit code is the return value of the invoked method.
+            object instance = null;
+            string log = $"RemoteExecuter: {assemblyName} {methodName} {string.Join(", ", additionalArgs)}{Environment.NewLine}";
+
+            try
+            {
+                // Create the test class if necessary
+                Assembly a = Assembly.Load(new AssemblyName(assemblyName));
+                Type t = a.GetType(typeName);
+                MethodInfo mi = t.GetTypeInfo().GetDeclaredMethod(methodName);
+                if (!mi.IsStatic)
+                {
+                    instance = Activator.CreateInstance(t);
+                }
+
+                // Invoke the test
+                object result = mi.Invoke(instance, additionalArgs);
+                returnData["Results"] = result is Task<int> task ? task.GetAwaiter().GetResult() : (int) result;
+            }
+            catch (Exception exc)
+            {
+                returnData["Results"] = -4;
+                log = log + exc;
+            }
+            finally
+            {
+                (instance as IDisposable)?.Dispose();
+            }
+            returnData["Log"] = log;
+        }
+
+        private bool GetStringValueFromValueSet(ValueSet message, string key, out string val)
+        {
+            val = null;
+            
+            object valObject;
+            if (!message.TryGetValue(key, out valObject))
```

Comment:
This will be case sensitive. I don't think we need to complicate it to support case insensitive as we'll have limited cases communicating between the service and the client.